client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java

```diff
@@ -522,11 +527,156 @@ protected void set(K key, V value) {
    }
 
    @Override
-   public Set<K> keySet() {
-      assertRemoteCacheManagerIsStarted();
-      // Use default scope
-      BulkGetKeysOperation<K> op = operationsFactory.newBulkGetKeysOperation(0);
-      return Collections.unmodifiableSet(op.execute());
+   public CloseableIteratorSet<K> keySet() {
+      return new KeySet();
+   }
+
+   @Override
+   public CloseableIteratorSet<Entry<K, V>> entrySet() {
+      return new EntrySet();
+   }
+
+   @Override
+   public CloseableIteratorCollection<V> values() {
+      return new ValuesCollection();
+   }
+
+   private class KeySet extends AbstractCollection<K> implements CloseableIteratorSet<K> {
+
+      @Override
+      public CloseableIterator<K> iterator() {
+         return new RemovableCloseableIterator<>(
+               new CloseableIteratorMapper<>(retrieveEntries(null, batchSize), e -> (K) e.getKey()),
```

Comment:
Sadly you can only supply a `KeyValueFilterConverter` which doesn't allow for null values. So I guess I will have it just return an empty byte[]. I will add a converter called `to_empty_bytes`. Seem alright to you guys?